diff --git a/leaflet.module b/leaflet.module
index e72a3e9..aa9ba3c 100644
--- a/leaflet.module
+++ b/leaflet.module
@@ -121,14 +121,16 @@ function leaflet_build_map($map, $features = array(), $height = '400px') {
   );
 
   // Load the leaflet library, which includes integration files.
-  libraries_load('leaflet');
+  $build['#attached']['libraries_load'][] = array(
+    'leaflet'
+  );
 
   return $build;
 }
 
 /**
  * DEPRECATED. Use leaflet_build_map() instead.
- * 
+ *
  * Load all Leaflet required client files and return markup for a map.
  *
  * @param array $map
