diff --git a/phone.info b/phone.info
index f398135..aa228e4 100644
--- a/phone.info
+++ b/phone.info
@@ -3,6 +3,7 @@ description = The phone module allows administrators to define a field type for
 package = Fields
 dependencies[] = field
 files[] = phone.migrate.inc
+files[] = phone.migrate-handler.inc
 files[] = tests/phone.au.test
 files[] = tests/phone.be.test
 files[] = tests/phone.br.test
diff --git a/phone.migrate-handler.inc b/phone.migrate-handler.inc
new file mode 100644
index 0000000..223a974 100644
--- /dev/null
+++ b/phone.migrate-handler.inc
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * @file
+ * MigratePhoneFieldHandler class definition.
+ */
+
+/**
+ * Wrap Migrate's simple field handler for 'phone' fields.
+ */
+class MigratePhoneFieldHandler extends MigrateSimpleFieldHandler {
+
+  public function __construct() {
+    $this->registerTypes(array('phone'));
+  }
+
+}
diff --git a/phone.migrate.inc b/phone.migrate.inc
index 2523ce0..488573f 100644
--- a/phone.migrate.inc
+++ b/phone.migrate.inc
@@ -27,14 +27,3 @@ function phone_field_alter(&$field_value, $instance_value) {
 function phone_migrate_api() {
   return array('api' => 2);
 }
-
-/**
- * Wrap Migrate's simple field handler for 'phone' fields.
- */
-class MigratePhoneFieldHandler extends MigrateSimpleFieldHandler {
-
-  public function __construct() {
-    $this->registerTypes(array('phone'));
-  }
-
-}
